util/stopper.go

```diff
@@ -162,13 +162,20 @@ func (s *Stopper) SetStopped() {
 	}
 }
 
-// Quiesce moves the stopper to state draining, waits until all tasks
-// complete, then moves back to non-draining state. This is used from
+// Quiesce moves the stopper to state draining. Unquiesce() needs to
+// be called before the stopper is stopped. This is used from
 // unittests.
 func (s *Stopper) Quiesce() {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	s.draining = 1
+	atomic.StoreInt32(&s.draining, 1)
```

Comment:
Hmm, updated, but I think `Quiesce()` still doesn't work in the following case:

0) `s.draining` is initially `0`.
1) `StartTask()` is called. It checks `s.draining`, but has not yet called `s.drain.Add(1)`.
2) `Quiesce()` is called. It sets `s.draining` and calls `s.drain.Wait()`. It finishes immediately.
3) `StartTask()` calls `s.drain.Add(1)`.
4) `Quiesce()` is called before all tasks have been drained.